third_party/move/tools/move-unit-test/src/lib.rs

```diff
@@ -156,38 +161,59 @@ impl UnitTestingConfig {
-            if self.ignore_compile_warnings {
-                Severity::NonblockingError
-            } else {
-                Severity::Warning
-            },
-        ) {
-            diagnostics::report_diagnostics(&files, diags);
-        }
+        if get_move_compiler_v2_from_env() {
+            let options = move_compiler_v2::Options {
+                compile_test_code: true,
+                testing: true,
+                sources: source_files,
+                dependencies: deps,
+                compiler_version: Some(CompilerVersion::latest_stable()),
+                language_version: Some(LanguageVersion::latest_stable()),
+                named_address_mapping: addresses
+                    .iter()
+                    .map(|(string, num_addr)| format!("{}={}", string, num_addr))
+                    .collect(),
+                ..Default::default()
+            };
+            let (files, units, opt_env) = build_and_report_v2_driver(options).unwrap();
+            let env = opt_env.expect("v2 driver should return env");
+            let test_plan = plan_builder_v2::construct_test_plan(&env, None);
+            test_plan.map(|tests| TestPlan::new(tests, files, units))
+        } else {
+            let (files, comments_and_compiler_res) = Compiler::from_files(
+                source_files,
+                deps,
+                addresses,
+                Flags::testing().set_skip_attribute_checks(false),
+                KnownAttribute::get_all_attribute_names(),
+            )
+            .run::<PASS_CFGIR>()
+            .unwrap();
+            let (_, compiler) =
+                diagnostics::unwrap_or_report_diagnostics(&files, comments_and_compiler_res);
+
+            let (mut compiler, cfgir) = compiler.into_ast();
+            let compilation_env = compiler.compilation_env();
+            let test_plan =
+                unit_test::plan_builder::construct_test_plan(compilation_env, None, &cfgir);
+
+            if let Err(diags) = compilation_env.check_diags_at_or_above_severity(
+                if self.ignore_compile_warnings {
+                    Severity::NonblockingError
+                } else {
+                    Severity::Warning
+                },
+            ) {
+                diagnostics::report_diagnostics(&files, diags);
+            }
 
-        let compilation_result = compiler.at_cfgir(cfgir).build();
+            let compilation_result = compiler.at_cfgir(cfgir).build();
 
-        let (units, warnings) =
-            diagnostics::unwrap_or_report_diagnostics(&files, compilation_result);
-        diagnostics::report_warnings(&files, warnings);
-        test_plan.map(|tests| TestPlan::new(tests, files, units))
+            let (units, warnings) =
+                diagnostics::unwrap_or_report_diagnostics(&files, compilation_result);
+            diagnostics::report_warnings(&files, warnings);
+            test_plan.map(|tests| TestPlan::new(tests, files, units))
```

Comment:
Last line is common for the 2 branches.  It might be better to wrap the whole `if` in a `let (test_plan, files, units) = if ...` and have a single such line.  I'm actually a little surprised we don't have more in common.

Maybe we didn't before, but we also should bail out if there are errors.